services/src/main/java/io/grpc/services/BinaryLogSinkProvider.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.services;
+
+import com.google.protobuf.MessageLite;
+import java.io.Closeable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Subclasses must be thread safe, and are responsible for writing the binary log message to
+ * the appropriate destination.
+ */
+@ThreadSafe
+public abstract class BinaryLogSinkProvider implements Closeable {
+
+  public static BinaryLogSinkProvider provider() {
```

Comment:
what does this method do?

Reply:
I have a PR to create a ServiceProvider util class. If it gets merged then we can avoid some duplicate code here.
https://github.com/grpc/grpc-java/pull/3886

Added a TODO placeholder.

Reply:
This was a mistake we made with ServerBuilder.   Having the static factory of the class be on itself and be extensible made for confusing behavior.    The factory and the Sink should be in separate classes.